app/services/jira_board_service.rb

```diff
@@ -0,0 +1,91 @@
+class JiraBoardService
+  attr_accessor :jira_board
+
+  def update!(params)
+    @jira_board = JiraBoard.find_by(id: params['id'])
+    jira_board.update!(update_params(params))
+    update_environments(params)
+    jira_board
+  end
+
+  def create!(params)
+    @jira_board = JiraBoard.create!(update_params(params))
+    update_environments(params)
+    jira_board
+  end
+
+  private
+
+  def update_jira_board(params)
+    jira_board.jira_project_key = params['jira_project_key']
+    jira_board.environment_field = params['environment_field']
+    jira_board.project_name = params['project_name']
+    jira_board.product_id = params['product_id']
```

Comment:
I don't like this approach that much. I'd like the use of a STRUCT, so you send an object here with the information you need, and not more. https://ruby-doc.org/core-2.5.0/Struct.html